### DISCUSSION TITLE
How to assign auth method to each listener?

### DISCUSSION THREAD

**Discussion by EgHubs** [emqx/emqx#11600] [Q&A]
i have two listners
normal ssl:8883 one which users connect to with MongoDB auth
and the other one is wss:8084 which I want a super user to join with a database file auth
The thing is that both are linked right now which I don't want
I mean whoever connects to 8883 with the username/password of the file which is the super user then he can join
How to achieve that the file is only accessible by 8084 port wss listener?
or else how can I achieve it in another way 
I tried to add a new user to the acl.conf file to allow him to publish messages to all 
`{allow, {username, "id3"}, publish, ["#"]}.

so he is not a super user but still anyone can access it 
Also I tried to connect for 8084 to fail if no peer cert 
but how can I achieve something like that
`{allow, {port, "8084"}, publish, ["#"]}. `

**Comment by qzhuyan** [comment]
maybe EMQX should support ACL rule taken zone as input.

You could also write a ACL plugin to achieve that.

**Reply by EgHubs** [reply]
Arent there any other way to  assign an authentication method to each listener
I have two methods how a listner can only access one authentication method and the other access only that other one?
